tgui/packages/tgui/interfaces/Fabricator.tsx

```diff
@@ -0,0 +1,337 @@
+                    </Tabs.Tab>
+
+                    {Object.values(namedCategories).map((categoryName) => (
+                      <Tabs.Tab
+                        key={categoryName}
+                        selected={selectedCategory === categoryName}
+                        onClick={() => {
+                          setSelectedCategory(categoryName);
+                          setSearchText('');
+                        }}
+                        fluid
+                        color="transparent">
+                        {categoryName} ({categoryCounts[categoryName]})
+                      </Tabs.Tab>
+                    ))}
+                  </Tabs>
+                </Section>
+              </Stack.Item>
+              <Stack.Item grow>
+                <Section
+                  title="Recipes"
+                  fill
+                  buttons={
+                    <Fragment>
+                      <Button.Checkbox
+                        onClick={() => setDisplayMatCost(!displayMatCost)}
+                        checked={displayMatCost}>
+                        Display Material Costs
+                      </Button.Checkbox>
+                      <Button
+                        content="R&D Sync"
+                        onClick={() => act('sync_rnd')}
+                      />
+                    </Fragment>
+                  }>
+                  <Stack vertical fill>
+                    <Stack.Item>
+                      <Section>
+                        <SearchBar />
+                      </Section>
+                    </Stack.Item>
+                    <Stack.Item grow>
+                      <Section fill scrollable>
+                        {Object.values(sortedDesigns)
+                          .filter(
+                            (design) =>
+                              selectedCategory === '__ALL' ||
+                              design.categories?.indexOf(selectedCategory) !==
+                                -1
+                          )
+                          .filter(
+                            (design) =>
+                              design.name.toLowerCase().indexOf(searchText) !==
+                              -1
+                          )
+                          .map((design) => (
+                            <Recipe
+                              key={design.name}
+                              design={design}
+                              available={availableMaterials}
+                            />
+                          ))}
+                      </Section>
+                      {busy ? (
+                        <Dimmer
+                          style={{
+                            'font-size': '2em',
+                            'text-align': 'center',
+                          }}>
+                          <Icon name="cog" spin />
+                          {' Building items...'}
+                        </Dimmer>
+                      ) : undefined}
```

Comment:
Change to `{busy && (<Dimmer>etc</Dimmer>)}`

Reply:
Using `&&` causes tgui to render a `0` here. I've had the same experience with other React-like frameworks.

![image](https://user-images.githubusercontent.com/12202230/185810104-2c2c5c73-cddf-4ea2-b4fd-6ecf92d1264a.png)

Thoughts?

Reply:
Nevermind, doing `!!<variable> && <component>` makes this work.

Reply:
It renders a zero because the evaluation of

`<number> && <component>`

will either return a number (zero) or the component, javascript doesn't convert the statement to boolean, otherwise it would not render the component as it would have been converted to true.

Adding to this, React frameworks do not render booleans but they do render numbers, so that's why you see the zero.